FirebaseStorageSwift/Sources/AsyncAwait.swift

```diff
@@ -0,0 +1,26 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import FirebaseStorage
+
+@available(iOS 15.0, *)
+public extension StorageReference {
+  @discardableResult
+  func putDataAwait(_ uploadData: Data,
+                    metadata: StorageMetadata? = nil) async throws -> StorageMetadata {
+    // TODO: Add a parameter to capture StorageUploadTask and to enable Progress tracking.
+//    -> StorageUploadTask {
+    return try await __putDataGlue(uploadData, metadata: metadata)
```

Comment:
With using continuations it will be something like:
```
@available(iOS 15.0, *)
  func data(maxSize: Int64) async throws -> Data {

    // Let's define a specific error for internal inconsistency. 
    // Alternatively we can do the same as we do in `getResultCallback` method.
    enum DataError: Error {
      case internalInconsistency
    }

    typealias DataContinuation = CheckedContinuation<Data, Error>
    return try await withCheckedThrowingContinuation({ (continuation: DataContinuation) in
      // TODO: Use task to handle progress and cancellation.
      let task = self.getData(maxSize: maxSize) { data, error in
        if let data = data {
          continuation.resume(returning: data)
        } else {
          continuation.resume(throwing: error ?? DataError.internalInconsistency)
        }
      }
    })
  }
```

Reply:
This worked really well. Thanks @maksymmalyhin !